CDP4Composition/Services/AppSettingService/AppSettings.cs

```diff
@@ -0,0 +1,34 @@
+﻿// -------------------------------------------------------------------------------------------------
+// <copyright file="AppSettings.cs" company="RHEA System S.A.">
+//   Copyright (c) 2018-2020 RHEA System S.A.
+// </copyright>
+// -------------------------------------------------------------------------------------------------
+
+namespace CDP4Composition.Services.AppSettingService
+{
+    using Newtonsoft.Json;
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// Base class from which all <see cref="AppSettings"/> shall derive
+    /// </summary>
+    public abstract class AppSettings
+    {
+        /// <summary>
+        /// Initializes a the list <see cref="Init"/> class
+        /// </summary>
+        public void Init(bool initializeDefaults)
+        {
+            if (initializeDefaults)
+            {
+                this.DisabledPlugins = new List<string>();
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the <see cref="DisabledPlugins"/>
+        /// </summary>
+        [JsonProperty]
+        public List<string> DisabledPlugins { get; set; }
```

Comment:
i am a bit surprised this is a list of string. I would have thought something a bit less "free" would be in order, like a dedicated class that has something like a `Name` and `IsEnabled` property? 

I guess this is an abstract class, because for every app (IME and AddIn) we need a dedicated class? Does this mean we only should have one type per app and not multiple?

Reply:
For now I told @mmilitaru to keep it simple and we will deal with this in the next task (which is really about the plugin disabling). For now, this is suitable to test the concept.

Reply:
And yes the idea is to have seperate classes for IME and Addin